flexmeasures/api/v3_0/sensors.py

```diff
@@ -1575,13 +1622,170 @@ def trigger_forecast(self, id: int, sensor: Sensor, **params):
+                        items:
+                          type: number
+                        description: Forecast values.
+                  examples:
+                    finished:
+                      summary: Finished forecasting job
+                      value:
+                        start: "2025-10-15T00:00:00+01:00"
+                        end: "2025-10-15T04:00:00+01:00"
+                        resolution: "PT1H"
+                        unit: "kW"
+                        values: [1.2, 1.5, 1.4, 0.8]
+            401:
+              description: UNAUTHORIZED
+            403:
+              description: INVALID_SENDER
+            422:
+              description: UNPROCESSABLE_ENTITY
+          tags:
+            - Sensors
+        """
+
+        try:
+            # Fetch job from RQ
+            queue = current_app.queues["forecasting"]
+            job = Job.fetch(job_id, connection=queue.connection)
+
+            # Job does not exist
+            if job is None:
+                return unprocessable_entity(f"Job {job_id} not found.")
+
+            # Ensure the sensor in the URL matches the sensor associated with the forecast job
+            job_sensor_id = job.meta.get("sensor_id")
+            if job_sensor_id != sensor.id:
+                return unprocessable_entity(
+                    f"Sensor ID mismatch: job {job_id} is for sensor {job_sensor_id}, not sensor {sensor.id}."
+                )
+            if job.dependency_ids:
+                return unprocessable_entity(
+                    f"Job {job_id} is the wrap-up job; please query one of those cycle jobs {job.dependency_ids} instead."
+                )
+
+            # Map RQ statuses to API statuses
+            status = job.get_status()
+            if status in ("queued", "started", "deferred"):
+                # Still running
+                response = dict(
+                    status="RUNNING" if status == "started" else "PENDING",
+                    job_id=job_id,
+                )
+                d, s = request_processed()
+                return dict(**response), s
+
+            if status == "failed":
+                # Return error message
+                response = dict(
+                    status="FAILED",
+                    job_id=job_id,
+                    error=str(job.exc_info),
+                )
+                d, s = request_processed()
+                return dict(**response), s
+
+            # Job finished → fetch forecasts from DB
+            # search for forecasts linked to this job and sensor
+            data_source = get_data_source_for_job(job, type="forecasting")
+
+            forecasts = sensor.search_beliefs(
+                event_starts_after=job.meta.get("start_predict_date"),
+                event_ends_before=job.meta.get("end_date") + sensor.event_resolution,
+                source=data_source,
+                most_recent_beliefs_only=True,
+                use_latest_version_per_event=True,
```

Comment:
I think you'll also need `one_deterministic_belief_per_event=True,` (just like in `get_schedule`), otherwise there is no guarantee to have 1 row per event (in case of multiple sources or probabilistic data).